### PR TITLE
Fix rename of VisibleDatasetFactory

### PR DESCRIPTION
**Pull request by ThibaudDauce**
Following https://github.com/opendatateam/udata/pull/2997

### CODE REVIEW

File: udata_front/tests/views/test_organization.py [hunk 1]
```diff
@@ -284,7 +276,6 @@ def test_resources_csv(self):
             for _ in range(3)
         ]
         not_org_dataset = DatasetFactory(resources=[ResourceFactory()])
-        hidden_dataset = DatasetFactory()
```
Comment (maudetes) [inline udata_front/tests/views/test_organization.py:287]: We now have a `HiddenDatasetFactory` with private=True if you wanna keep tests on hidden datasets?

File: udata_front/theme/gouvfr/templates/dataset/display.html [hunk 2]
```diff
@@ -15,7 +15,7 @@
     'description': dataset.description|mdstrip(60)|forceescape,
     'image': dataset.organization and dataset.organization.logo(external=True) or '',
     'keywords': [_('dataset')] + dataset.tags,
-    'robots': 'noindex, nofollow' if dataset.is_hidden or 'inspire:identifier' in dataset.extras else '',
+    'robots': 'noindex, nofollow' if not dataset.resources or 'inspire:identifier' in dataset.extras else '',
```
Comment (maudetes) [inline udata_front/theme/gouvfr/templates/dataset/display.html:18]: We should actually keep on using `is_hidden`, where the logic is.

**Review by maudetes — COMMENTED** [review]
Thank you for this fix! :pray:
I have minor comments on tests

File: udata_front/tests/views/test_organization.py (hunk 1 above)
Comment (ThibaudDauce) [reply udata_front/tests/views/test_organization.py:287]: I'm not sure I'm a big fan of `HiddenDatasetFactory`. I start using it but is it really useful to abstract away the fact that it's just `private=True`?

Comment (ThibaudDauce) [reply udata_front/tests/views/test_organization.py:287]: And yes you're right I need to add back the test :-) (didn't understood your comment)

File: udata_front/theme/gouvfr/templates/dataset/display.html (hunk 2 above)
Comment (maudetes) [reply udata_front/theme/gouvfr/templates/dataset/display.html]: If indexable by data.gouv.fr, it should probably be indexable and used for crawling by search engine?

Comment (ThibaudDauce) [reply udata_front/theme/gouvfr/templates/dataset/display.html]: Done in f70fb40f

File: udata_front/tests/views/test_organization.py (hunk 1 above)
Comment (ThibaudDauce) [reply udata_front/tests/views/test_organization.py:287]: Done in 5241907a

File: udata_front/tests/views/test_dataset.py [hunk 3]
```diff
@@ -153,14 +152,6 @@ def test_200_if_deleted_but_authorized(self):
         response = self.get(url_for('datasets.show', dataset=dataset))
         self.assert200(response)
 
-    def test_no_index_on_empty(self):
```
Comment (maudetes) [inline udata_front/tests/views/test_dataset.py:156]: Maybe, we could rebrand it to `test_no_index_on_archived`? With an archived dataset as example?

Comment (ThibaudDauce) [reply udata_front/tests/views/test_dataset.py:156]: Done in 4fc7cf61

**Review by maudetes — APPROVED** [review]
Thank you for the fix! :pray:

**Review by nicolaskempf57 — APPROVED** [review]

**Pull request merged**